demo/site/src/blocks/YouTubeVideoBlock.tsx

```diff
@@ -13,19 +13,23 @@ const getHeightInPercentForAspectRatio = (aspectRatio: YouTubeVideoBlockData["as
     }
 };
 
+const EXPECTED_YT_ID_LENGTH = 11;
+
+const parseYoutubeIdentifier = (value: string): string | undefined => {
+    // regex from https://stackoverflow.com/a/51870158
+    const regExp =
+        /(https?:\/\/)?(((m|www)\.)?(youtube(-nocookie)?|youtube.googleapis)\.com.*(v\/|v=|vi=|vi\/|e\/|embed\/|user\/.*\/u\/\d+\/)|youtu\.be\/)([_0-9a-zA-Z-]+)/;
+    const match = value.match(regExp);
+    const youtubeId = value.length === EXPECTED_YT_ID_LENGTH ? value : match && match[8].length == EXPECTED_YT_ID_LENGTH ? match[8] : null;
```

Comment:
From the provided StackOverflow answer: "ID will always be in match group 8". Why do you use match group 9?

Reply:
Maybe they confused match count and index or something. As you can see the match at index 8 is in fact the id.

![image](https://github.com/vivid-planet/comet/assets/56400587/366d0532-9e44-41a0-abc6-bafbe3000817)